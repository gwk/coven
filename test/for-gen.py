from sys import argv


'''
These tests exercised FOR_ITER cases that were tricky when coven was first written for python 3.7.
Their efficacy has yet to be evaluated for 3.10.
'''

def dummy(): pass # add non-trivial loop body and return to make sure that StopIteration gets the right line.

def iter_multi():
  for i in range(1):
    dummy()
  dummy()

def iter_inline():
  for i in range(1): dummy()

def iter_trivial():
  for i in range(1): pass

def gen_multi():
  for i in (j for j in range(1)):
    dummy()
  dummy()

def gen_inline():
  for i in (j for j in range(1)): dummy()

def gen_trivial():
  for i in (j for j in range(1)): pass

iter_multi()
iter_inline()
iter_trivial()

gen_multi()
gen_inline()
gen_trivial()
