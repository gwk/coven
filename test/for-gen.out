----------------
Coverage Report:

__main__: for-gen.py: 26 traceable; 25 covered; 0 ignored; 0 ignored but covered; 0 not covered.
